finder.py

```diff
@@ -322,10 +322,26 @@ def find_match(slug):
 		timestamp = datetime.datetime.strptime(clip_info["vod_data"]["created_at"], '%Y-%m-%dT%H:%M:%SZ')
 		timestamp = int(timestamp.replace(tzinfo=datetime.timezone.utc).timestamp())
 		timestamp += clip_info["vod"]["offset"]
-		new_diff = (timestamp - match["start_time"]) // 60
+		new_diff = (timestamp - best_match["start_time"]) // 60
 		if new_diff > 0:
 			result["better_minutes_diff"] = new_diff
 
+	# Get league clip
+	match_details_url = f"https://api.opendota.com/api/matches/{best_match['match_id']}"
+	try:
+		response = requests.get(match_details_url)
```

Comment:
To reduce the number of calls, we should cache this request similar to what we're doing in `retrieve_clip_info()`. You could add a function similar to this:
```
def retrieve_match_info(match_id):
	filename = cache_filename(f"match_{match_id}", "json")
	if os.path.exists(filename):
		with open(filename, "r") as f:
			return json.loads(f.read())
	else:
		match_details_url = f"https://api.opendota.com/api/matches/{match_id}"
		# maybe add try/catch block here for JSONDecoder exception
		response = requests.get(match_details_url)
		if response.status_code != 200:
			return None
		data = response.json()
		with open(filename, "w+") as f:
			f.write(json.dumps(data, indent="\t"))
		return data
```

Reply:
Added with 904317c